scalding-repl/src/main/scala/com/twitter/scalding/ReplImplicits.scala

```diff
@@ -77,14 +77,15 @@ object ReplImplicits extends FieldConversions {
   /* Using getter/setter here lets us get the correct defaults set by the mode
      (and read from command-line, etc) while still allowing the user to customize it */
   def config: Config = Config.defaultFrom(mode) ++ customConfig
-  def config_=(c: Config) { customConfig = c }
+
+  protected def shell: BaseScaldingShell = ScaldingShell
```

Comment:
seems strange to have this called BaseReplImplicits. The implicit part is about the imports. What is the core function here? If we are refactoring, I'd like to improve it, seems we make be making it worse here.